avr-hal-generic/src/port.rs

```diff
@@ -207,11 +207,12 @@ impl<PIN: PinOps, MODE: mode::Io> Pin<MODE, PIN> {
         ADC: crate::adc::AdcOps<H>,
         CLOCK: crate::clock::Clock,
     {
-        let new = Pin {
+        let mut new = Pin {
             pin: self.pin,
             _mode: PhantomData,
         };
         adc.enable_pin(&new);
+        unsafe { new.pin.make_input(false) };
```

Comment:
Thanks so much for digging into this.  I agree that this is the sensible thing to do and arguably, it should have always been this way.